### PR TITLE
Queries for complexity metrics

### PR DESCRIPTION
**Pull request by fabsx00**
I am bringing in examples of queries, and I'm making them unit tests so they continue to work. This PR brings in queries for complexity metrics.

### CODE REVIEW

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala [hunk 1]
```diff
@@ -18,4 +18,19 @@ class AstNodeMethods(val node: nodes.AstNode) extends AnyVal {
     * */
   def astMinusRoot: AstNode = node.start.astMinusRoot
 
+  def isControlStructure: Boolean = node.start.isControlStructure.size == 1
```
Comment (Ferada) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala]: I, what?

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala [hunk 2]
```diff
@@ -18,4 +18,19 @@ class AstNodeMethods(val node: nodes.AstNode) extends AnyVal {
...
+
+  def depth(p: nodes.AstNode => Boolean = _ => true): Int = {
```
Comment (Ferada) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala]: Also these should be commented I guess.

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala [hunk 3]
```diff
@@ -18,4 +18,19 @@ class AstNodeMethods(val node: nodes.AstNode) extends AnyVal {
...
+
```
Comment (Ferada) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala]:
```suggestion
    node.start.astChildren.map(_.depth(p)).l.fold(if (p(node)) { 1 } else { 0 })(Math.max)
```
At least the last line can be a fold instead.

Comment (fabsx00) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala]: As discussed offline, leaving it as-is for now.

**Pull request merged**